src/frontend/hooks/useHentMinimalFagsak.ts

```diff
@@ -0,0 +1,51 @@
+import { useQuery } from '@tanstack/react-query';
+
+import { useHttp } from '@navikt/familie-http';
+
+import { hentMinimalFagsak } from '../api/hentMinimalFagsak';
+import { useAppContext } from '../context/AppContext';
+import type { IMinimalFagsak } from '../typer/fagsak';
+import { PersonType } from '../typer/person';
+import { sammenlignFødselsdato } from '../utils/obfuskerData';
+
+function obfuskerFagsak(fagsak: IMinimalFagsak) {
+    fagsak.gjeldendeUtbetalingsperioder.forEach(gup => {
+        let indeks = 1;
+        gup.utbetalingsperiodeDetaljer.sort(sammenlignFødselsdato).forEach(upd => {
+            if (upd.person.type === PersonType.SØKER) {
+                upd.person.navn = 'Søker Søkersen';
+            } else {
+                upd.person.navn = '[' + indeks++ + '] Barn Barnesen';
+            }
+        });
+    });
+}
+
+export const HentMinimalFagsakQueryKeyFactory = {
+    fagsak: (fagsakId: number | undefined) => ['minimalfagsak', fagsakId],
+};
+
+export function useHentMinimalFagsak(
+    fagsakId: number | undefined,
+    påvirkerSystemLaster: boolean = true
+) {
+    const { request } = useHttp();
+    const { skalObfuskereData } = useAppContext();
+    return useQuery({
+        queryKey: HentMinimalFagsakQueryKeyFactory.fagsak(fagsakId),
+        queryFn: async () => {
+            if (fagsakId === undefined) {
+                return Promise.reject(new Error('Kan ikke hente fagsak uten fagsakId.'));
+            }
+            const fagsak = await hentMinimalFagsak(request, fagsakId, påvirkerSystemLaster);
+            return Promise.resolve(fagsak);
+        },
```

Comment:
Tror dette kan potensielt forenkles til noe ala:

```suggestion
        queryFn: () => {
            if (fagsakId === undefined) {
                return Promise.reject(new Error('Kan ikke hente fagsak uten fagsakId.'));
            }
            return hentMinimalFagsak(request, fagsakId, påvirkerSystemLaster);
        },
```

Du får vurdere om det er noe du har lyst til å gjøre

Reply:
Hvordan funker dette, vil ikke hooken da returnere en Promise etterom hentMinimalFagsak returnerer en Promise?